core/src/main/kotlin/knote/PageManagerImpl.kt

```diff
@@ -219,10 +213,19 @@ internal class PageManagerImpl(
         file.writeText(content)
     }
 
-    private var watchJob: Job? = null
-    private fun startWatcher() {
-        NotebookManagerImpl.logger.debug("starting page watcher")
-        watchJob = watchActor(notebookScript.pageRoot.absoluteFile.toPath()) {
+    override fun watchDataFile(pageId: String, file: File) {
+        val page = pages[pageId] as? PageImpl ?: run {
+            throw IllegalStateException("page $pageId cannot be loaded as PageImpl")
```

Comment:
interesting - I didn't know you could kind of execute actions with run with the null check.  I've been dying to find an alternative since they took out the tenary operator